lam/lib/modules/takUser.inc

```diff
@@ -0,0 +1,270 @@
+			],
+            [
+				'name' => 'takUser_takRole',
+				'description' => _('TAK team role'),
+				'help' => 'takRole',
+                'default' => 'Team Member',
+				'values' => 'Team Member, Team Leader, HQ, Sniper, Medic, Forward Observer, RTO, K9',
+				'example' => _('Team Member'),
+                'required' => true
+			],
+            [
+				'name' => 'takUser_takColor',
+				'description' => _('TAK team color'),
+				'help' => 'takColor',
+                'default' => 'Cyan',
+				'values' => 'Blue, Brown, Cyan, Dark Blue, Dark Green, Green, Magenta, Maroon, Orange, Purple, Red, Teal, White, Yellow',
+				'example' => _('Cyan'),
+                'required' => true
+			],
+		];
+		// available PDF fields
+		$return['PDF_fields'] = [
+			'takCallsign' => _('TAK callsign'),
+			'takRole' => _('TAK team role'),
+			'takColor' => _('TAK team color'),
+		];
+		// profile options
+		$profileContainer = new htmlResponsiveRow();
+		$profileContainer->add(new htmlResponsiveInputField(_('TAK Callsign'), 'takUser_takCallsign', null, 'takUserCallsign'), 12);
+		$profileContainer->add(new htmlResponsiveInputField(_('TAK Team Role'), 'takUser_takRole', null, 'takUserRole'), 12);
+		$profileContainer->add(new htmlResponsiveInputField(_('TAK Team Color'), 'takUser_takColor', null, 'takUserColor'), 12);
+		$return['profile_options'] = $profileContainer;
+		/**
+		* not working: will not allow -
+		* // profile checks
+		* $return['profile_checks']['takUser_takCallsign'] = [
+		*	'type' => 'ext_preg',
+		*	'regex' => 'ascii',
+		*	'error_message' => $this->messages['takCallsign'][0]];
+		*/
+		return $return;
+	}
+
+	/**
+	 * This function fills the $messages variable with output messages from this module.
+	 */
+	function load_Messages() {
+		$this->messages['takCallsign'][0] = ['ERROR', _('User name'), _('User name contains invalid characters. Valid characters are: a-z, A-Z, 0-9 and .-_')];
+		$this->messages['takCallsign'][1] = ['ERROR', _('Please enter a unique callsign.')];
+		$this->messages['takCallsign'][2] = ['ERROR', _('A TAK login with this callsign already exists. Please choose another callsign.')];
+		$this->messages['takRole'][1] = ['ERROR', _('Role'), _('Please select a role.')];
+		$this->messages['takColor'][1] = ['ERROR', _('Please select a team color.')];
+		$this->messages['takUsers'][0] = ['ERROR', _('takUsers'), _('Unable to add the user to the takUsers group')];
+	}
+
+	/**
+	 * Returns the HTML meta data for the main account page.
+	 *
+	 * @return array HTML meta data
+	 */
+	function display_html_attributes() {
+		$return = new htmlResponsiveRow();
+		// Section title
+		$return->add(new htmlSubTitle(_('General')));
+		// takCallsign
+		$this->addSimpleInputTextField($return, 'takCallsign', _('TAK callsign'), true);
+		// takRole
+		$return->add(new htmlResponsiveSelect('takRole', ['Team Member', 'Team Leader', 'HQ', 'Sniper', 'Medic', 'Forward Observer', 'RTO', 'K9'], ['Team Member'], _('TAK team role'), 'takRole'), 12);
```

Comment:
Please define the possible roles+colors in one central place (and use it for e.g. file upload, too).